### PR TITLE
Issue #1184 Better error message on clustering service start failures

### PR DESCRIPTION
**Pull request by chrisdennis**
I've gone with what I understood to be the accepted terminology here "server storage manager".  That is I think the most important thing to review here.

### CODE REVIEW

**Comment by AbfrmBlr** [comment]
LGTM

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java [hunk 1]
```diff
@@ -132,12 +132,19 @@ public void start(final ServiceProvider<Service> serviceProvider) {
       try {
         entity = entityFactory.retrieve(entityIdentifier, configuration.getServerConfiguration());
       } catch (EntityNotFoundException e) {
-        /*
-         * If the connection failed because of a creation failure, re-throw the creation failure.
-         */
-        throw new IllegalStateException(failure == null ? e : failure);
+        if (failure == null) {
+          throw new IllegalStateException("The server-side storage manager does not exist."
```
Comment (albinsuresh) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java]: The entity is defined as _Ehcache clustered entity_ in our clustered cache doc. So I guess that's what we should be using here.

**Comment by albinsuresh** [comment]
I see that the word _Clustered Store Manager_ is used all over the place in the `EhcacheActiveEntity` class. Either that should change or the term that is documented should change. I'm not saying that it should be in this PR. May be another one.

**Comment by ljacomet** [comment]
I vote against using `Store` in any user documentation. That term has way too much meaning associated with it. So my preference goes to _Ehcache clustered entity_.

**Comment by ljacomet** [comment]
Still believe this adds something after the rework from @lorban so good to go once rebased

**Comment by lorban** [comment]
Agreed; this change makes some errors more intelligible. Let's leave terminology to #1243.

**Pull request merged**